advanced/Scripts/piholeLogFlush.sh

```diff
@@ -8,17 +8,38 @@
 # This file is copyright under the latest version of the EUPL.
 # Please see LICENSE file for your rights under this license.
 
-echo -n "::: Flushing /var/log/pihole.log ..."
-# Test if logrotate is available on this system
-if command -v /usr/sbin/logrotate >/dev/null; then
-  # Flush twice to move all data out of sight of FTL
-  /usr/sbin/logrotate --force /etc/pihole/logrotate; sleep 3
-  /usr/sbin/logrotate --force /etc/pihole/logrotate
+if [[ "$@" != *"quiet"* ]]; then
+  echo -n "::: Flushing /var/log/pihole.log ..."
+fi
+if [[ "$@" == *"once"* ]]; then
+  # Nightly logrotation
+  if command -v /usr/sbin/logrotate >/dev/null; then
+    # Logrotate once
+    /usr/sbin/logrotate --force /etc/pihole/logrotate
+  else
+    # Copy pihole.log over to pihole.log.1
+    # and empty out pihole.log
+    # Note that moving the file is not an option, as
+    # dnsmasq would happily continue writing into the
+    # moved file (it will have the same file handler)
+    cp /var/log/pihole.log /var/log/pihole.log.1
+    echo " " > /var/log/pihole.log
```

Comment:
Will this cause a space to be put in front of the first log line, possibly messing up the parsing of the first line? However, I do see that it's the original behavior.